front/lib/api/signup.ts

```diff
@@ -335,17 +322,13 @@ export async function handleRegularSignupFlow(
       origin: "auto-joined",
     });
 
-    return new Ok({ flow: null, workspace });
-  } else if (targetWorkspaceId && !canJoinTargetWorkspace) {
```

Comment:
that test was invalid -- `canJoinTargetWorkspace` is a function, so `!canJoinTargetWorkspace` is always false.